### PR TITLE
fix: Update lock and remove python limit fo pylate and colbert_engine

### PR DESCRIPTION
**Pull request by Samoed**
Remove `python_full_version < '3.13'` for pylate and colbert_engine. They're supporting higher python verisions. I updated `uv.lock`

### CODE REVIEW

File: tests/test_evaluators/test_ClusteringEvaluator.py [hunk 1]
```diff
@@ -42,4 +43,4 @@ def encode(  # noqa: PLR6301
         )
         result = clusterer(model, encode_kwargs={"batch_size": 32})
 
-        assert result == [2, 1, 0]
+        assert v_measure_score(labels, result) == 1.0
```
Comment (Samoed) [inline tests/test_evaluators/test_ClusteringEvaluator.py:46]: Order of elements was changed in recent scikit-learn version https://scikit-learn.org/stable/whats_new/v1.9.html#sklearn-cluster

**Comment by Samoed** [comment]
> Fix [metrics.accuracy_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html#sklearn.metrics.accuracy_score), [metrics.hamming_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.hamming_loss.html#sklearn.metrics.hamming_loss) [metrics.zero_one_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.zero_one_loss.html#sklearn.metrics.zero_one_loss), [metrics.matthews_corrcoef](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html#sklearn.metrics.matthews_corrcoef) and [metrics.confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix) (when labels is not None) now raise an error when y_true is string and y_pred is numeric, for all array-like inputs. Previously, lists and numpy arrays not of object dtype did not raise an error for this mixed input case. The above metrics will also raise an error for [label indicator matrix](https://scikit-learn.org/stable/glossary.html#term-label-indicator-matrix) inputs of inconsistent size, except for [metrics.confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix) which does not accept label indicator matrix inputs. By [Lucy Liu](https://github.com/lucyleeow). [#33086](https://github.com/scikit-learn/scikit-learn/pull/33086) https://scikit-learn.org/stable/whats_new/v1.9.html#sklearn-metrics

Breaks our zero-shot tasks. I always wanted to fix it to support more metrics, but it seems this is really necessary now.

**Review by KennethEnevoldsen — APPROVED** [review]
for now should we set an upper limit for sklearn until we have fixes the issues?

Otherwise good

**Comment by Samoed** [comment]
Created an issue https://github.com/embeddings-benchmark/mteb/issues/4784
